x/bank/keeper/keeper.go

```diff
@@ -0,0 +1,103 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	accountkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	distrkeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+// Copied from https://github.com/CudoVentures/cosmos-sdk/blob/3816012a2d4ea5c9bbb3d8e6174d3b96ff91a039/x/bank/keeper/keeper.go#L19C1-L20C1
+var burnDenom = "acudos"
+
+type Keeper struct {
+	bankkeeper.BaseKeeper
+
+	ak accountkeeper.AccountKeeper
+
+	dk    distrkeeper.Keeper
+	dkSet bool
+}
+
+var _ bankkeeper.Keeper = Keeper{}
+
+func NewCustomKeeper(
+	cdc codec.BinaryCodec,
+	storeKey storetypes.StoreKey,
+	ak accountkeeper.AccountKeeper,
+	paramSpace paramtypes.Subspace,
+	blockedAddrs map[string]bool,
+) Keeper {
+	keeper := Keeper{
+		BaseKeeper: bankkeeper.NewBaseKeeper(cdc, storeKey, ak, paramSpace, blockedAddrs),
+		ak:         ak,
+	}
+	return keeper
+}
+
+func (k *Keeper) SetDistrKeeper(dk distrkeeper.Keeper) {
+	k.dk = dk
+	k.dkSet = true
+}
+
+// Migrate from https://github.com/CudoVentures/cosmos-sdk/blob/3816012a2d4ea5c9bbb3d8e6174d3b96ff91a039/x/bank/keeper/keeper.go#L439
+func (k Keeper) BurnCoins(ctx sdk.Context, moduleName string, amounts sdk.Coins) error {
```

Comment:
Suggest starting with:

```go
if !k.dkSet {
  panic(sdkerrors.Wrapf(sdkerrors.ErrLogic, "distribution keeper not set"))
}
```

since we should never be able to get to BurnCoins with dk not being set.